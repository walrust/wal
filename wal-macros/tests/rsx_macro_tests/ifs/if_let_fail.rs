use wal_macros::rsx;

fn main() {
    rsx! { if let };
    rsx! { if let {} };
    rsx! { if let true {}};
    rsx! { if let {} {} };
}
