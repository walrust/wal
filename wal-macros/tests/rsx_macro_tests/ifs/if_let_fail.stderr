error: unexpected end of input, expected one of: identifier, `::`, `<`, `_`, literal, `const`, `ref`, `mut`, `&`, parentheses, square brackets, `..`, `const`
 --> tests/rsx_macro_tests/ifs/if_let_fail.rs:4:5
  |
4 |     rsx! { if let };
  |     ^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected one of: identifier, `::`, `<`, `_`, literal, `const`, `ref`, `mut`, `&`, parentheses, square brackets, `..`, `const`
 --> tests/rsx_macro_tests/ifs/if_let_fail.rs:5:20
  |
5 |     rsx! { if let {} };
  |                    ^

error: expected `=`
 --> tests/rsx_macro_tests/ifs/if_let_fail.rs:6:25
  |
6 |     rsx! { if let true {}};
  |                         ^

error: expected one of: identifier, `::`, `<`, `_`, literal, `const`, `ref`, `mut`, `&`, parentheses, square brackets, `..`, `const`
 --> tests/rsx_macro_tests/ifs/if_let_fail.rs:7:20
  |
7 |     rsx! { if let {} {} };
  |                    ^
